alodataset/merge_dataset.py

```diff
@@ -20,18 +20,35 @@ class MergeDataset(torch.utils.data.Dataset):
         List of datasets
     transform_fn : function
         transformation applied to each sample
+    weights: List[int] | None
+        For N datasets, a list of N integer weights.
+        The samples from a dataset with weight `w` will appear `w` times in the MergeDataset.
     """
 
-    def __init__(self, datasets, transform_fn=None):
+    def __init__(self, datasets, transform_fn=None, weights=None):
         self.datasets = datasets
+        self.weights = self._init_weights(weights)
         self.indices = self._init_indices()
         self.transform_fn = transform_fn
 
+    def _init_weights(self, weights):
+        n_datasets = len(self.datasets)
+        if weights is None:
+            return [1] * n_datasets
+
+        if len(weights) != n_datasets:
+            raise RuntimeError("The number of weights should be equal to the number of datasets.")
+
+        if any(type(w) != int for w in weights):
+            raise RuntimeError("weights should be a list of int.")
+        return weights
+
     def _init_indices(self):
         indices = []
         for dset_idx, dset in enumerate(self.datasets):
-            for idx in range(len(dset)):
-                indices.append((dset_idx, idx))
+            for _ in range(self.weights[dset_idx]):
```

Comment:
Not sure to understand how the following is doing importance sampling based on the weights.
What I understan id that the default scenario is that you have [1, 1] (two dataset). Let's say instead you have [2, 1] to give more importance to the first dataset. You code will do the same thing no matter the weights value.

Reply:
Let me know if I missed something